### DISCUSSION TITLE
Data validation issue with jsonschema

### DISCUSSION THREAD

**Discussion by vanand123** [python-jsonschema/jsonschema#1262] [Q&A] [answered]
Getting error while trying to validate data like below:
```
import json
from jsonschema import validate, ValidationError

schema = {
    "properties": {
        "Enabled": {
            "type": "string",
            "description": "Enable or disable Access.",
            "pattern": "^(?i)(on|off)$"
        }
    },
    "required": ["Enabled"],
    "additionalProperties": False,
    "type": ["object"]
}

data = '{"Enabled": "On"}'
data_dict = json.loads(data)

try:
    validate(instance=data_dict, schema=schema)
    print("Validation successful!")
except ValidationError as e:
    print(f"Validation error: {e.message}")
```
**Error:**
```
jsonschema.exceptions.SchemaError: '^(?i)(on|off)$' is not a 'regex'

Failed validating 'format' in metaschema['allOf'][1]['properties']['properties']['additionalProperties']['$dynamicRef']['allOf'][3]['properties']['pattern']:
    {'format': 'regex', 'type': 'string'}

On schema['properties']['Enabled']['pattern']:
    '^(?i)(on|off)$'
```

The above test code is working fine with "Python 3.8.18 & jsonschema-4.17.3". However, throws above error with "Python 3.11.9 & jsonschema-4.17.3" 
Is there any usage change [around case insensitive pattern (?i)] with respect to jsonschema for Python 3.11.x?

**Comment by Julian** [comment, marked as answer]
Nothing in this library, see https://docs.python.org/3/library/re.html, that indeed isn't a valid Python regex anymore, the `(^i)` must come at the beginning: 

> Changed in version 3.11: This construction can only be used at the start of the expression.

**Reply by vanand123** [reply]
Thanks @Julian for quick response. Updating pattern to **"pattern": "(?i)^(on|off)$"** worked fine.
Please proceed to close this discussion.
